eclipse-glsp/glsp-client
# Issues/1336

#### What it does

Introduce an optional grid module to deal with a grid layout

- Optional grid module defines a grid based on x/y coordinates
-- Adds the grid as a background that properly zooms/resizes
-- By default uses the GridSnapper for positioning elements
-- Use grid for helper lines and movement tools

- Optional debug module that shows the bounds of elements
-- Useful for debugging but not meant for production

- Render optional debug and grid toggles in tool palette if present

- Add both optional modules to the workflow example

Fixes https://github.com/eclipse-glsp/glsp/issues/1336

**_Please note this PR builds upon https://github.com/eclipse-glsp/glsp-client/pull/342 and should only be merged afterwards!_**

#### How to test

- See if everything still works as expected.
- Use the new grid feature in the workflow module.

#### Follow-ups

#### Changelog


If the optional accessibility module was used, the grid was derived through the `GridSnapper`, this needs to be replaced by a binding of the `Grid`.
